### PR TITLE
Mecanum drive

### PR DESCRIPTION
**Pull request by dixonwille**
@technomancers/developers @technomancers/devops

### CODE REVIEW

File: src/main/java/org/usfirst/frc/team1758/robot/commands/DriveWithJoystick.java [hunk 1]
```diff
@@ -7,26 +7,21 @@
 	private static boolean finished;
 	public DriveWithJoystick() {
 		requires(driveTrain);
-		requires(sensors);
+    requires(sensors);
```
Comment (dixonwille) [inline src/main/java/org/usfirst/frc/team1758/robot/commands/DriveWithJoystick.java]: This space is bothering me haha.

File: src/main/java/org/usfirst/frc/team1758/robot/commands/DriveWithJoystick.java [hunk 2]
```diff
@@ -7,26 +7,21 @@
...
 	}
 	protected void execute() {
 		//If you need to do calculations on the Axis do them in a new method inside the Controller
-		double x = OI.drivingController.getNormalizedAxis(Axes.LEFT_X);
-		double y = OI.drivingController.getNormalizedAxis(Axes.LEFT_Y);
-		double twist = OI.drivingController.getTwist();
-		double gyroAngle = sensors.getGyroAngle();
-		driveTrain.mecanumDrive(x, y, twist, gyroAngle);
+	  double gyroAngle = sensors.getGyroAngle();
+		driveTrain.mecanumDrive(OI.drivingController.getNormalizedAxis(Axes.LEFT_X), OI.drivingController.getNormalizedAxis(Axes.LEFT_Y), OI.drivingController.getNormalizedAxis(Axes.RIGHT_X), gyroAngle);
```
Comment (dixonwille) [inline src/main/java/org/usfirst/frc/team1758/robot/commands/DriveWithJoystick.java]: I know this is my bad but for readability, let us pull out the Axes into their own variables.

File: src/main/java/org/usfirst/frc/team1758/robot/subsystems/DriveTrain.java [hunk 3]
```diff
@@ -3,46 +3,27 @@
 import org.usfirst.frc.team1758.robot.RobotMap;
 import org.usfirst.frc.team1758.robot.commands.DriveWithJoystick;
 
+import com.ctre.CANTalon;
+
+
```
Comment (dixonwille) [inline src/main/java/org/usfirst/frc/team1758/robot/subsystems/DriveTrain.java]: Try and keep multiple spaces out of the project. Single space is fine but more than that is a lot of extra white space.

File: src/main/java/org/usfirst/frc/team1758/robot/subsystems/DriveTrain.java [hunk 4]
```diff
@@ -3,46 +3,27 @@
...
 import edu.wpi.first.wpilibj.RobotDrive;
 
-import edu.wpi.first.wpilibj.Talon;
 import edu.wpi.first.wpilibj.command.Subsystem;
 
 public class DriveTrain extends Subsystem {
-	private Talon rf_Motor, rm_Motor, rb_Motor, lf_Motor, lm_Motor, lb_Motor;
+	private CANTalon rf_Motor, rm_Motor, rb_Motor, lf_Motor, lm_Motor, lb_Motor;
```
Comment (dixonwille) [inline src/main/java/org/usfirst/frc/team1758/robot/subsystems/DriveTrain.java]: We are not using `rm_Motor` and `lm_Motor` so we should remove them.

File: src/main/java/org/usfirst/frc/team1758/utilities/Controller.java [hunk 5]
```diff
@@ -6,9 +6,10 @@
 public class Controller
 {
 	private Joystick joystick;
+	private final double threshold = 0.1;
```
Comment (dixonwille) [inline src/main/java/org/usfirst/frc/team1758/utilities/Controller.java]: We should make this changable and not static. Best way to do that is to have a parameter that has a default value in the Constructor method. Again I know I did this.

File: src/main/java/org/usfirst/frc/team1758/robot/commands/CenterReflectiveTape.java [hunk 6]
```diff
@@ -1,24 +1,24 @@
...
+// package org.usfirst.frc.team1758.robot.commands;
 
-public class CenterReflectiveTape extends CommandBase
-{
-	private boolean finished;
-	public CenterReflectiveTape()
-	{
-		requires(vision);
-	}
+// public class CenterReflectiveTape extends CommandBase
```
Comment (dixonwille) [inline src/main/java/org/usfirst/frc/team1758/robot/commands/CenterReflectiveTape.java]: This class is completley empty. May be worthwhile just to delete it.

**Review by dixonwille — COMMENTED** [review]
So I think the next steps are to:

* Tune the talons (more research into this)
* Install cameras and play around with them a bit more.

The latter only if the build team does not have anything for us yet.

**Review by EtashGuha — APPROVED** [review]

**Pull request merged**